pyteal/ast/assert_.py

```diff
@@ -12,22 +13,56 @@
 class Assert(Expr):
     """A control flow expression to verify that a condition is true."""
 
-    def __init__(self, cond: Expr, *additional_conds: Expr) -> None:
+    def __init__(
+        self,
+        cond: Expr | tuple[Expr, Optional[str]],
+        *additional_conds: Expr | tuple[Expr, Optional[str]]
```

Comment:
Alternative interface idea: what about adding a keyword arg like this?

```python
def __init__(self, cond: Expr, *additional_conds: Expr, *, comment: str) -> None:
```

A benefit of this is that I think it'd be easier for the developer to add/remove these as needed when debugging, and it's very explicit.

A downside is that multi-clause Asserts aren't as well supported. However, I argue this isn't as important, since the caller can always refactor `Assert(x, y, z)` into `Seq(Assert(x, comment="..."), Assert(y, z))` if they want to really target their comments.